gdlsc.go

```diff
@@ -1,104 +1,182 @@
 package main
 
 import (
-	"encoding/json"
 	"fmt"
 	"io/ioutil"
-	"log"
-	"net/http"
 	"os"
-	"os/exec"
+	"path"
+	"path/filepath"
+	"regexp"
 	"strings"
+
+	"sort"
+	"strconv"
+
+	"github.com/ryanuber/go-license"
 )
 
-// URL represents path a github repos license
-var URL = "https://api.github.com/repos/%s/license?access_token=" + os.Getenv("ACCESS_TOKEN")
+const root = "vendor/"
 
-//types for message to be printed
-type message struct {
-	Dependency string
-	License    string
-	Link       string
+// Response contains license and file info
+type response struct {
+	License string
+	File    string
 }
 
-// options for response from github
-type response struct {
-	Html_URL string
-	License  license
+// ByName type is used to implement sort interface to sort response by name
+type ByName []response
+
+func (s ByName) Len() int           { return len(s) }
+func (s ByName) Less(i, j int) bool { return s[i].File < s[j].File }
+func (s ByName) Swap(i, j int)      { s[i], s[j] = s[j], s[i] }
+
+func (r response) String() string {
+	return r.File + " ------- > " + r.License + "\n"
 }
 
-type license struct {
-	Name string
+type request struct {
+	Path        string
+	LicenseFile string
 }
 
-// uses github api to fetch license based on link generated
-func getLicense(dependency string, ch chan message) {
-	url := formatLink(dependency)
-	resp, err := http.Get(url)
+func formatLicense(file string, size int) string {
+	f, err := ioutil.ReadFile(file)
 	if err != nil {
 		panic(err)
 	}
+	data := strings.Split(string(f), "\n")
+	if size <= len(data) {
+		data = data[0:size]
+	}
+	formatted := "UNKOWN LICENSE\n\n|\t" + strings.Join(data, "\n|\t") + "\n"
+	return formatted
+}
 
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
+func isWtf(file string) bool {
+	match := "everyone is permitted to copy and distribute"
```

Comment:
I think it would be more robust to actually match the header of the license:
        DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE 
                    Version 2, December 2004